backend/main/server/resources/Archive.py

```diff
@@ -0,0 +1,134 @@
+
+class ArchiveListResource(Resource):
+    @cache.cached(timeout=100)
+    def get(self, who):
+        """Gets all archive links"""
+        archiveTable = tableConversion(who)
+        if archiveTable is None:
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+
+        archives = archiveTable.query.all()
+        archives = archives_schema.dump(archives)
+
+        if not archives:
+            return {'status': 'success', 'archives': archives}, 206  # Partial Content Served
+
+        return {'status': 'success', 'archives': archives}, 200
+
+    @jwt_required()
+    def post(self, who):
+        """Add archive link"""
+        archiveTable = tableConversion(who)
+        if archiveTable is None:
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+
+        json_data = request.get_json(force=True)
+
+        if not json_data:
+            return {'status': 'fail', 'message': 'No input data'}, 400
+
+        errors = archive_schema.validate(json_data)
+
+        if errors:
+            return {'status': 'fail', 'message': 'Error handling request'}, 422
+
+        data = archive_schema.load(json_data)
+
+        archive = archiveTable.query.filter_by(archiveURL=data.get('archiveURL')).first()
+
+        if archive:
+            return {'status': 'fail', 'message': 'archive already exists'}, 400
+
+        archive = archive(archiveURL=data.get('archiveURL'))
+
+        db.session.add(archive)
+        db.session.commit()
+
+        return {'status': 'success', 'message': 'archive successfully created'}, 201
+
+
+class ArchiveResource(Resource):
+    @cache.cached(timeout=100)
+    def get(self, who, archiveID):
+        """Get an archive by archive ID"""
+        archiveTable = tableConversion(who)
+        if archiveTable is None:
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+
+        archive = archiveTable.query.filter_by(archiveID=archiveID)
+
+        if not archive.first():
+            return {'status': 'fail', 'message': 'No archive with ID ' + str(archiveID) + ' exists'}, 404
+
+        archive = archives_schema.dump(archive)
+        return {'status': 'success', 'archives': archive}, 200
+
+    @jwt_required()
+    def delete(self, who, archiveID):
+        """Delete an archive by ID"""
+        archiveTable = tableConversion(who)
+        if archiveTable is None:
+            return {'status': 'fail', 'message': 'No data for ' + str(who) + ' exists'}, 404
+
+        archive = archiveTable.query.filter_by(archiveID=archiveID)
+
+        if not archive.first():
+            return {'status': 'fail', 'message': 'No archive with ID ' + str(archiveID) + ' exists'}, 404
+
+        archive.delete()
+        db.session.commit()
+        return {'status': 'sucess', 'message': 'archive Deleted'}, 200
+
+
+class ArchiveRandomResource(Resource):
+    def get(self, who):
+        """Get a random archive link"""
```

Comment:
Nit: Maybe put a note here about how the database will be the source of random by entry order. Code might confuse someone later.

Reply:
Got it